docs/how-tos/hello-world-module.md

```diff
@@ -390,13 +494,197 @@ You don't need to edit any of the validate or configuration methods because you'
+
+   ```go {class="line-numbers linkable-line-numbers" data-start="6"}
+   "os"
+   "reflect"
+   "io/ioutil"
+   ```
+
+1. Add `imagePath = ""` to the global variables so you have the following:
+
+   ```go {class="line-numbers linkable-line-numbers" data-line="22" data-start="19" data-line-offset="19"}
+   var (
+       HelloCamera      = resource.NewModel("jessamy", "hello-world", "hello-camera")
+       errUnimplemented = errors.New("unimplemented")
+       imagePath        = ""
+   )
+   ```
+
+1. In the test script you hard-coded the path to the image.
+   For the module, let's make the path a configurable attribute so you or other users of the module can set the path from which to get the image.
+
+   Edit the `type Config struct` definition, replacing the comments with the following:
+
+   ```go {class="line-numbers" data-start="33"}
+   type Config struct {
+       resource.AlwaysRebuild
+       ImagePath string `json:"image_path"`
+   }
+   ```
+
+   This adds the `image_path` attribute and causes the resource to rebuild each time the configuration is changed.
+
+1. Delete the `Reconfigure()` function entirely, since the camera will rebuild instead of reconfiguring.
+
+1. We are not providing a default image but rely on the end user to supply a valid path to an image when configuring the resource.
+   This means `image_path` is a required attribute.
+   Add the following code to the `Validate` function to throw an error if `image_path` isn't configured or isn't a string:
+
+   ```go {class="line-numbers linkable-line-numbers" data-start="38"}
+   func (cfg *Config) Validate(path string) ([]string, error) {
+     var deps []string
+     if cfg.ImagePath == "" {
+         return nil, resource.NewConfigValidationFieldRequiredError(path, "image_path")
+     }
+     if reflect.TypeOf(cfg.ImagePath).Kind() != reflect.String {
+         return nil, errors.New("image_path must be a string.")
+     }
+     imagePath = cfg.ImagePath
+     return deps, nil
+   }
+   ```
+
+1. The module generator created a stub for the `Image` function we want to implement:
+
+   ```go {class="line-numbers linkable-line-numbers" data-start="103" }
+   func (s *helloWorldHelloCamera) Image(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
+       panic("not implemented")
+   }
+   ```
+
+   You need to replace `panic("not implemented")` with code to actually implement the method:
+
+   ```go {class="line-numbers linkable-line-numbers" data-start="104" }
+   imgFile, err := os.Open(imagePath)
+   if err != nil {
+     return nil, camera.ImageMetadata{}, errors.New("Error opening image.")
+   }
+   defer imgFile.Close()
+   imgByte, err := ioutil.ReadFile(imagePath)
+   return imgByte, camera.ImageMetadata{}, nil
+   ```
+
+1. Delete the `SubscribeRTP` and `Unsubscribe` methods, since they are not applicable to this camera.
```

Comment:
does it matter if we leave them in?

Reply:
It's marginally easier to delete than copy paste more lines

Reply:
```suggestion
1. Delete the `SubscribeRTP` and `Unsubscribe` methods. They are not applicable to this camera.
```